java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSinkFactory.java

```diff
@@ -89,21 +85,19 @@ public void validate(
             }
         }
 
+        RestHighLevelClientAdapter client;
         // 2. check connection
-        RestClientBuilder builder = RestClient.builder(host);
-        if (config.getPassword() != null && config.getUsername() != null) {
-            final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
-            credentialsProvider.setCredentials(
-                    AuthScope.ANY,
-                    new UsernamePasswordCredentials(config.getUsername(), config.getPassword()));
-            builder.setHttpClientConfigCallback(
-                    httpClientBuilder ->
-                            httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider));
-        }
-        RestHighLevelClient client = new RestHighLevelClient(builder);
-        // Test connection
         try {
-            boolean isConnected = client.ping(RequestOptions.DEFAULT);
+            boolean isConnected;
+            if (config.getConnector().equals("elasticsearch")) {
+                client = new ElasticRestHighLevelClientAdapter(host, config);
+                isConnected = client.ping(RequestOptions.DEFAULT);
```

Comment:
It seems that here is the only usage of `ping`. 

Here can know the concrete type of `client`, and therefore we don't have to make `ping` a method in interface. We can just make `ping` a normal method of the two clients and make the method parameter type to be the concrete type, to avoid the dirty type casting.